omegaconf/basecontainer.py

```diff
@@ -330,14 +330,24 @@ def expand(node: Container) -> None:
                         dest._validate_merge(key=key, value=src_value)
                         dest_node._merge_with(src_value)
                     else:
-                        dest.__setitem__(key, src_value)
+                        if src_value != MISSING:
+                            dest.__setitem__(key, src_value)
                 else:
                     if isinstance(src_value, BaseContainer):
-                        dest.__setitem__(key, src_value)
+                        if src_value != MISSING:
```

Comment:
> Yes.

I'm not entirely convinced because in your example `a` is the container, but `a != MISSING`. Anyway, resolving since you removed that check.